app/models/protocol.rb

```diff
@@ -508,6 +508,20 @@ def has_non_first_draft_ssrs?
     sub_service_requests.where.not(status: 'first_draft').any?
   end
 
+  def has_incomplete_additional_details?
+    if service_requests.any? { |sr| sr.additional_detail_services.present? }
+      service_requests.each do |sr|
```

Comment:
I think it would be a good idea to get rid of this nested loop. Consider adding `has_many :line_items, through: :service_requests` to the Protocol model. Then you could just write something like `line_items.any?(&:has_incomplete_additional_details?)`.